### ISSUE TITLE
sql: flake in TestLogic/.../schema_change_in_txn/rollback_mutations

### ISSUE THREAD

**Issue by knz** [cockroachdb/cockroach#80378]
Found in https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests_BazelUnitTests/4969598?showRootCauses=false&expandBuildChangesSection=true&expandBuildDeploymentsSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true

```
    logic.go:2837:
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7071/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-dbg/bin/pkg/sql/logictest/logictest_test_/logictest_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/schema_change_in_txn:726:
        expected:
        violates unique constraint
        got:
        pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_SERIALIZABLE - failed preemptive refresh due to a conflict: committed value on key /Table/15/1/755513646351286273/1/1): "sql txn" meta={id=bd351dee key=/Table/3/1/138/2/1 pri=0.00639744 epo=0 ts=1650634901.028145957,1 min=1650634900.519720866,0 seq=67} lock=true stat=PENDING rts=1650634900.519720866,0 wto=false gul=1650634901.019720866,0
        HINT: See: https://www.cockroachlabs.com/docs/dev/transaction-retry-error-reference.html#retry_serializable
    logic.go:2477:
         /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7071/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-dbg/bin/pkg/sql/logictest/logictest_test_/logictest_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/schema_change_in_txn:729: too many errors encountered, skipping the rest of the input
[13:41:41] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7071/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-dbg/bin/pkg/sql/logictest/logictest_test_/logictest_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/schema_change_in_txn with config fakedist-spec-planning: 191 tests, 2 failures
[13:41:41] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7071/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-dbg/bin/pkg/sql/logictest/logictest_test_/logictest_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/save_table: 16 statements
            --- FAIL: TestLogic/fakedist-spec-planning/schema_change_in_txn/rollback_mutations (0.72s)
```

Jira issue: CRDB-15739

**Comment by ajwerner** [comment]
Relates to https://github.com/cockroachdb/cockroach/issues/52366. We don't currently have a way to mark transactions as retriable in logic tests.

**Comment by rafiss** [comment]
resolved by 28c48f9b5d8eade07934c7a894a98aa5a790347d
